### ISSUE TITLE
Scheduled function deploy doesn't report the error in non-debug run

### ISSUE THREAD

**Issue by kika** [firebase/firebase-tools#3259]
### [REQUIRED] Environment info

**firebase-tools:** `9.8.0`

**Platform:** MacOS

### [REQUIRED] Test case

```typescript
import * as functions from 'firebase-functions';

export const scheduledBalanceUpdate =
  functions.pubsub.schedule('every 1 minute').onRun(async (_) => {
    log.debug(`Balance updater run`);
    const start = Date.now();
    // await payment.balanceUpdateAll();
    log.debug(`Balance updater finished: ${Date.now() - start} milliseconds`);
  });

```

Note the error: it's `minute` not `minutes` as per spec.

### [REQUIRED] Steps to reproduce

`firebase deploy --only functions`

```
✔  functions: Finished running predeploy script.
i  functions: ensuring required API cloudfunctions.googleapis.com is enabled...
i  functions: ensuring required API cloudbuild.googleapis.com is enabled...
✔  functions: required API cloudbuild.googleapis.com is enabled
✔  functions: required API cloudfunctions.googleapis.com is enabled
i  functions: preparing . directory for uploading...
i  functions: packaged . (299 KB) for uploading
✔  functions: . folder uploaded successfully
i  functions: updating Node.js 10 function scheduledBalanceUpdate(us-central1)...
i  scheduler: ensuring required API cloudscheduler.googleapis.com is enabled...
i  pubsub: ensuring required API pubsub.googleapis.com is enabled...
✔  scheduler: required API cloudscheduler.googleapis.com is enabled
✔  pubsub: required API pubsub.googleapis.com is enabled
✔  functions: created scheduler job firebase-schedule-scheduledBalanceUpdate-us-central1
✔  functions[scheduledBalanceUpdate(us-central1)]: Successful update operation.
✔  Deploy complete!
```

Everything appears to be just fine.

### [REQUIRED] Expected behavior

Scheduler job is not created and error is given.

### [REQUIRED] Actual behavior

Scheduler job appears to be created, but it's not. The only way to reveal the error is to run with `--debug` flag, see below.

```
[2021-04-05T00:38:51.900Z] 	Error during upsert schedule for projects/cuida-app/locations/us-central1/functions/scheduledBalanceUpdate: Failed to create scheduler job projects/cuida-app/locations/us-central1/jobs/firebase-schedule-scheduledBalanceUpdate-us-central1: HTTP Error: 400, Schedule or time zone is invalid.

Error: Functions did not deploy properly.
```

**Comment by Sisekelo** [comment]
did anyone get a solution here?

**Comment by nok91** [comment]
I had a similar issue:
[2021-12-31T13:47:11.314Z] Error: Failed to upsert schedule function alertHandler in region us-central1...

I've been able to resolve it by changing the Default GCP resource location from the Project settings of your project under the General Tab.

**Comment by Lyokone** [comment]
Hey 👋 This is fixed in the latest firebase tools version (12.7.0). You are now getting a proper issue:
```
Failed to create scheduler job projects/firebase-tools-test/locations/us-central1/jobs/firebase-schedule-scheduledBalanceUpdate-us-central1: HTTP Error: 400, Schedule or time zone is invalid.

Functions deploy had errors with the following functions:
        scheduledBalanceUpdate(us-central1)
```
